lib/phoenix_mtm/mappers.ex

```diff
@@ -0,0 +1,72 @@
+defmodule PhoenixMTM.Mappers do
+  @moduledoc ~S"""
+  A collection of commonly used mappers for the `collection_checkboxes` helper.
+
+  To use, pass a capture of the mapping function you wish to use to the
+  `collection_checkboxes` helper.
+
+  ## Example
+      <%= PhoenixMTM.Helpers.collection_checkboxes f, :tags,
+            Enum.map(@tags, &({&1.name, &1.id})),
+            selected: Enum.map(f.data.tags, &(&1.id)),
+            mapper: &PhoenixMTM.Mappers.nested/6
+
+  ## Using Custom Mappers
+
+  If you want to make your own custom mapper, you can optionally
+  `use PhoenixMTM.Mappers` and bring in some of the Phoenix tag helpers.
+
+  This is not required, as you can manually include which ever imports you want.
+  """
+
+  import Phoenix.HTML.Form
+  import Phoenix.HTML.Tag
+
+  @doc ~S"""
+  Checkbox input and label returned as a 2 element list - the default.
+
+  ### Example Output
+
+  ```html
+  <input type="checkbox" value="1" name="checkbox_1">
+  <label for="checkbox_1">1</label>
+  ```
+  """
+  def unwrapped(form, field, input_opts, label_text, label_opts, _opts) do
+    [
+      tag(:input, input_opts),
+      label(form, field, "#{label_text}", label_opts)
+    ]
+  end
+
+  @doc ~S"""
+  Checkbox input and label returned as a label with the checkbox and label text
+  nested within.
+
+  ### Example Output
+
+  ```html
+  <label for="checkbox_1">
+    <input type="checkbox" value="1" name="checkbox_1">
+    1
+  </label>
+  ```
+  """
+  def nested(form, field, input_opts, label_text, label_opts, _opts) do
+    [
+      label(form, field, label_opts) do
+        [
+          tag(:input, input_opts),
+          {:safe, "#{label_text}"}
```

Comment:
My tests give different results. Please see the following:

```elixir
iex(whatever@127.0.0.1)1> Phoenix.HTML.html_escape("<nasty>stuff</nasty>")
{:safe, "&lt;nasty&gt;stuff&lt;/nasty&gt;"}

iex(whatever@127.0.0.1)2> Phoenix.HTML.html_escape({:safe, "<nasty>stuff</nasty>"})
{:safe, "<nasty>stuff</nasty>"}

iex(whatever@127.0.0.1)3> Phoenix.HTML.raw("<nasty>stuff</nasty>")
{:safe, "<nasty>stuff</nasty>"}

iex(whatever@127.0.0.1)4> Phoenix.HTML.html_escape(Phoenix.HTML.raw("<nasty>stuff</nasty>"))
{:safe, "<nasty>stuff</nasty>"}
```

In other words, it seems to me we are doing the manual equivalent of a call to `raw/1`, which would in fact mean that nothing gets escaped at all.

If I was to choose, I would certainly `html_escape/1` the label texts. The use-cases where this isn't desired are exactly those that benefit from a custom `mapper`.